src/Symfony/Component/PropertyInfo/WriteMutator.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\PropertyInfo;
+
+/**
+ * Writes mutator tell how to write to a property.
+ *
+ * @author Joel Wurtz <jwurtz@jolicode.com>
+ */
+final class WriteMutator
+{
+    public const TYPE_METHOD = 1;
+    public const TYPE_PROPERTY = 2;
+    public const TYPE_ARRAY_DIMENSION = 3;
+    public const TYPE_ADDER_REMOVER_METHOD = 4;
+    public const TYPE_CONSTRUCTOR = 5;
+
+    private $type;
+    private $name;
+    private $adderName;
+    private $removerName;
+    private $private;
+    private $constructorParameter;
+    private $static;
+
+    public function __construct(int $type, ?string $name, ?string $adderName, ?string $removerName, bool $private = false, bool $static = false, \ReflectionParameter $constructorParameter = null)
+    {
+        $this->type = $type;
+        $this->name = $name;
+        $this->adderName = $adderName;
+        $this->removerName = $removerName;
+        $this->private = $private;
+        $this->constructorParameter = $constructorParameter;
+        $this->static = $static;
+    }
+
+    public function getType(): int
+    {
+        return $this->type;
+    }
+
+    public function getName(): ?string
+    {
+        return $this->name;
+    }
+
+    public function getAdderName(): ?string
+    {
+        return $this->adderName;
+    }
+
+    public function getRemoverName(): ?string
+    {
+        return $this->removerName;
+    }
+
+    public function isPrivate(): bool
+    {
+        return $this->private;
+    }
+
+    public function isStatic(): bool
+    {
+        return $this->static;
+    }
+
+    public function getConstructorParameter(): ?\ReflectionParameter
```

Comment:
This is what i'm using in the AutoMapper component and when it rewrite this i was wondering the same question, i am also in favor of not exposing the reflection parameter here as it's a detail of implementation, but need to rebase the AutoMapper component to see what info i need here.

Reply:
does a reflection parameter determine the position of the constructor argument? if it does or has some other relevant upside over the string name, i think its ok to have this here.
 
 if not, i would prefer a string, and if the AutoMapper needs to extract the reflection for BC it can do that iself, rather than have a new component do unnecessary things to make something else easier to allow BC.

Reply:
This has been remove on this PR, will readd it if needed, but i don't think it's useful in the Automapper